Node fails with: mmal_component_create_core: could not find component 'vc.ril.camera'

OS: Ubuntu Server 20.04.01 installed from the new official imager application
Platform: Raspberry Pi 4
ROS2 Foxy released binaries

Hi,

The node builds fine.
raspistill works, saves to file, shows preview...everything's fine.

When I run the node, I get the following:

```
[raspicam2_node-1] mmal: mmal_component_create_core: could not find component 'vc.ril.camera'
[raspicam2_node-1] mmal: Failed to create camera component
[ERROR] [raspicam2_node-1]: process has died [pid 2901, exit code -11, cmd XXXXXXXXXX'].
[DEBUG] [launch.launch_context]: emitting event: 'launch.events.process.ProcessExited'
[DEBUG] [launch]: processing event: '<launch.events.process.process_exited.ProcessExited object at 0xffff81a02d00>'
[DEBUG] [launch]: processing event: '<launch.events.process.process_exited.ProcessExited object at 0xffff81a02d00>' ✓ '<launch.event_handlers.on_process_exit.OnProcessExit object at 0xffff80c83b50>'
[raspicam2_node-1] Failed to create camera componentinit_cam: Failed to create camera componentTried to start capture before 

```
Every post I looked at on the subject refer to some badly connected or faulty sensor but the camera is fine with raspistill /raspivid apps.

Olivier

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,21 @@
 cmake_minimum_required(VERSION 3.5)
 project(raspicam2)
 
-set(CMAKE_CXX_STANDARD 14)
+# Default to C99
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+set (CMAKE_SHARED_LINKER_FLAGS "-Wl,--no-as-needed")
 
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)

--- a/src/RasPiCamPublisherNode.cpp
+++ b/src/RasPiCamPublisherNode.cpp
@@ -6,6 +6,42 @@ RasPiCamPublisher::RasPiCamPublisher(rclcpp::NodeOptions options)
 
     configure_parameters(*state);
 
+    //declare all parameters
+    declare_parameter("width");
+    declare_parameter("height");
+    declare_parameter("fps");
+    declare_parameter("quality");
+    declare_parameter("image_transport");
+    declare_parameter("enable_imv");
+    declare_parameter("camera_id");
+    declare_parameter("sharpness");
+    declare_parameter("contrast");
+    declare_parameter("brightness");
+    declare_parameter("saturation");
+    declare_parameter("ISO");
+    declare_parameter("videoStabilisation");
+    declare_parameter("exposureCompensation");
+    // declare_parameter("exposureMode");
+    // declare_parameter("flickerAvoidMode");
+    // declare_parameter("exposureMeterMode");
+    // declare_parameter("awbMode");
+    // declare_parameter("imageEffect");
+    declare_parameter("colourEffects_enable");
+    declare_parameter("colourEffects_u");
+    declare_parameter("colourEffects_v");
+    declare_parameter("rotation");
+    declare_parameter("hflip");
+    declare_parameter("vflip");
+    declare_parameter("roi_x");
+    declare_parameter("roi_y");
+    declare_parameter("roi_w");
+    declare_parameter("roi_h");
+    declare_parameter("shutter_speed");
+    declare_parameter("awb_gains_r");
+    declare_parameter("awb_gains_b");
+    declare_parameter("analog_gain");
+    declare_parameter("digital_gain");
+
     // get parameters
     int w, h, f, q;
     get_parameter_or("width", w, 320);

